frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/validation-utils.ts

```diff
@@ -256,8 +306,7 @@ const resourceDefinition = (formValues: PipelineBuilderFormYamlValues, taskType:
               taskType,
             );
           },
-        )
-        .required(i18n.t('pipelines-plugin~Required')),
+        ),
```

Comment:
One of the latest issues was that the workspace was always required, also if the resource is optional.

Instead of this "hard coded" required call, I added a `is-resource-is-required` check above.